src/solve.rs

```diff
@@ -191,6 +190,11 @@ pub async fn solve(
         })
     }
 
+    // 4a step: Update execution plan coordinates once we have all plans prepared
```

Comment:
I guess since setting the `internal` flag now happens within step 5 there will be no step 4b?
```suggestion
    // 4 step: Update execution plan coordinates once we have all plans prepared
```

Reply:
Hm, I've added 4a as a mid step, it might be confusing. So I updated the steps now.

Reply:
Ahh yes, of course 4 already existed as a step before. 🤦‍♂️ 
I think I was just to dense to get it while only looking at the diff (without seeing step 4). 😅 
Both variants now make sense to me.